src/overlays/actors/ovl_En_Death/z_en_death.c

```diff
@@ -50,230 +69,1690 @@ ActorInit En_Death_InitVars = {
+            for (effect = this->miniDeaths[i]->effects, j = 0; j < MINIDEATH_NUM_EFFECTS; j++, effect++) {
+                cmf->mf[3][0] = effect->pos.x;
+                cmf->mf[3][1] = effect->pos.y - 12.0f;
+                cmf->mf[3][2] = effect->pos.z;
+
+                gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 165, 255, 215, alpha);
+                gSPSegment(POLY_XLU_DISP++, 0x08,
+                           Gfx_TwoTexScroll(play->state.gfxCtx, 0, 0, 0, 32, 64, 1, 0,
+                                            ((play->gameplayFrames + ((i + j) * 10)) * -20) & 511, 32, 128));
+                gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(POLY_XLU_DISP++, gEffFire1DL);
+            }
+        }
+    }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void EnDeath_DrawCore(EnDeath* this, PlayState* play) {
+    s32 pad;
+    Gfx* gfx;
+
+    OPEN_DISPS(play->state.gfxCtx);
+    gfx = POLY_OPA_DISP;
+
+    Matrix_ReplaceRotation(&play->billboardMtxF);
+
+    gSPMatrix(&gfx[0], Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(&gfx[1], gGomessCoreDL);
+
+    if (this->actor.params >= 5) {
+        // bats in front of the core
+        gSPDisplayList(&gfx[2], gGomessBatsGuardingCoreDL);
+        POLY_OPA_DISP = &gfx[3];
+    } else {
+        POLY_OPA_DISP = &gfx[2];
+    }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+s32 EnDeath_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDeath* this = THIS;
+
+    if (this->noDrawLimbs[limbIndex] == true) {
+        *dList = NULL;
+    } else if (limbIndex == GOMESS_LIMB_CLOAK_LOWER && this->actionFunc == EnDeath_DeathCutscenePart3) {
+        if (this->actionTimer - 5 > 0) {
+            f32 scale = 1.0f - 0.1f * (this->actionTimer - 5);
+
+            Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+        }
+    }
+
+    if (limbIndex == GOMESS_LIMB_SCYTHE_HANDLE || limbIndex == GOMESS_LIMB_SCYTHE_BLADE) {
+        if (!this->holdsScythe) {
+            *dList = NULL;
+        }
+    } else if (limbIndex == GOMESS_LIMB_CLOAK_UPPER) {
+        rot->z += this->cloakUpperRotationModifier;
+    } else if (limbIndex == GOMESS_LIMB_CLOAK_LOWER) {
+        rot->z += this->cloakLowerRotationModifier;
+    }
+    return 0;
+}
+
+void EnDeath_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f sWeaponColliderOffsetEndSwing = { -1000.0f, 12000.0f, 0.0f };
+    static Vec3f sWeaponColliderOffsetOther = { -2000.0f, -2500.0f, 0.0f };
+    static Vec3f sDamageEffectOffsets[5] = {
+        { 1000.0f, 2000.0f, 0.0f },    { 1500.0f, 1000.0f, 4000.0f },  { 1500.0f, 1000.0f, -4000.0f },
+        { 4000.0f, 4000.0f, 2000.0f }, { 4000.0f, 4000.0f, -2000.0f },
+    };
+    static s8 sDamageEffectPosMap[GOMESS_LIMB_MAX] = {
```

Comment:
```suggestion
    static s8 sLimbToBodyParts[GOMESS_LIMB_MAX] = {
```
This actually looks like a LimbToBodyParts array